### DISCUSSION TITLE
aws-events: grantPutEventsTo() not working as expected

### DISCUSSION THREAD

**Discussion by jabalsad** [aws/aws-cdk#23164] [General]
### Describe the bug

Granting PutEvents access to the default EventBus does not result in a resource-based policy with the permissions.

```
        const eventBus = events.EventBus.fromEventBusName(this, "DefaultBus", "default");
        eventBus.grantPutEventsTo(new iam.AccountPrincipal("123456789"));
```

### Expected Behavior

Expect that default event bus has a resource policy with put permissions for the principal account.

### Current Behavior

Event bus does not have the relevant permissions policy.

### Reproduction Steps

```
        const eventBus = events.EventBus.fromEventBusName(this, "DefaultBus", "default");
        eventBus.grantPutEventsTo(new iam.AccountPrincipal("123456789"));
```

### CDK CLI Version

2.51.0

### Node.js Version

16.14.2

### OS

Mac OSX

### Language

Typescript

**Comment by peterwoodworth** [comment]
This method works by adding a statement to the policy of whichever principal you want to have these permissions. Since account principals don't have policy documents, this is a no-op

If you need to accomplish this, you may need to [drop down to the L1](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_events.CfnEventBusPolicy.html) `CfnEventBusPolicy`

Converting to a discussion, as that will be a better place for you to get assistance in finding a workaround if necessary

**Reply by jabalsad** [reply]
Thanks! I'll give the L1 construct try. I would add that perhaps the documentation could describe that limitation.

**Reply by peterwoodworth** [reply]
I think a warning might also be effective. We'll keep this feedback in mind 🙂

**Reply by prli** [reply]
I found [this](https://github.com/aws/aws-cdk/issues/19034) which contains useful information and examples.
